### PR TITLE
Add utility to format ranges

### PR DESCRIPTION
**Pull request by K-ballo**

### CODE REVIEW

**Review by msimberg — DISMISSED** [review]
Looks good to me. Did you already have a specific use-case in mind (and if yes, I guess that will be changed separately)?

**Comment by K-ballo** [comment]
>  Did you already have a specific use-case in mind

I had some when I first wrote this, but I'm not sure those are still present today. Another PR which transitions more stuff from iostreams to format is in the works (branch `more-format`)

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

File: libs/core/format/include/hpx/modules/format.hpp [hunk 1]
```diff
@@ -276,4 +276,36 @@ namespace hpx { namespace util {
...
+                    using value_type = std::decay_t<decltype(elem)>;
+                    detail::formatter<value_type>::call(os, spec, &elem);
+                }
+            }
+        };
+    }    // namespace detail
+
+    template <typename Range>
+    detail::format_join<Range> format_join(
+        Range const& range, boost::string_ref delimiter = "") noexcept
```
Comment (msimberg) [inline libs/core/format/include/hpx/modules/format.hpp:307]: Absolutely not a blocker, but would `","` be a more appropriate default delimiter? Or not having a default at all?

Comment (K-ballo) [reply libs/core/format/include/hpx/modules/format.hpp:307]: > would `","` be a more appropriate default delimiter?
no

> Or not having a default at all?
sure, go ahead and remove the default

**Pull request merged**